File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
import sys

input = sys.stdin.readline

n, m = map(int, input().split())
nums = list(map(int, input().split()))

s, e = 0, 0
sum = nums[0]
cnt = 0

while True:
    if sum == m:
        cnt += 1

    if sum >= m:
        s += 1
        sum -= nums[s - 1]
    else:
        if e == n-1:
            break
        e += 1
        sum += nums[e]

print(cnt)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int s = 0;
        int e = 0;
        int sum = 0;
        int cnt = 0;

        while (true) {
            if (sum >= m) {
                sum -= arr[s++];
            } else if (e == n) {
                break;
            } else {
                sum += arr[e++];
            }
            if (sum == m) {
                cnt++;
            }
        }

        bw.write(Integer.toString(cnt));
        bw.close();

    }
}
